src/js/disability-benefits/526EZ/components/ReviewCardField.jsx

```diff
@@ -0,0 +1,189 @@
+    required: false,
+    disabled: false,
+    readonly: false,
+  }
+
+
+  constructor(props) {
+    super(props);
+
+    // Throw an error if there’s no viewComponent (should be React component)
+    if (typeof get('ui:options.viewComponent', this.props.uiSchema) !== 'function') {
+      throw new Error(`No viewComponent found in uiSchema for ReviewCardField ${this.props.idSchema.$id}.`);
+    }
+
+    if (!['object', 'array'].includes(this.props.schema.type)) {
+      throw new Error(`Unknown schema type in ReviewCardField: ${this.props.schema.type}`);
+    }
+
+    this.state = {
+      // Set initial state based on whether all the data is valid
+      editing: !errorSchemaIsValid(props.errorSchema)
+    };
+  }
+
+  onPropertyChange(name) {
+    return (value) => {
+      const formData = Object.keys(this.props.formData || {}).length
+        ? this.props.formData
+        : getDefaultFormState(this.props.schema, undefined, this.props.registry.definitions);
+      this.props.onChange(set(name, value, formData));
+    };
+  }
+
+
+  /**
+   * Much of this is taken from ArrayField & ObjectField.
+   *
+   * Renders a SchemaField for each property it wraps.
+   */
+  getEditView = () => {
+    const {
+      disabled,
+      errorSchema,
+      formData,
+      idSchema,
+      onBlur,
+      readonly,
+      registry,
+      schema,
+      uiSchema
+    } = this.props;
+    const { SchemaField } = registry.fields;
+
+    let properties;
+    if (schema.type === 'object') {
+      properties = Object.keys(schema.properties);
+    } else if (schema.type === 'array') {
+      properties = Object.keys(schema.items.properties);
+    } else {
+      // Shouldn't get here, but in case we update the gate in the constructor later...
+      throw new Error(`Unknown schema type in ReviewCardField: ${schema.type}`);
+    }
+
+    const title = this.getTitle();
+
+    return (
+      <div className="review-card">
+        <div className="review-card--body input-section va-growable-background">
+          <h4>{title}</h4>
+          {properties.map(propName => {
+            return (
+              <div key={propName}>
+                <SchemaField
+                  name={propName}
+                  required={this.isRequired(propName)}
+                  schema={schema.properties[propName]}
+                  uiSchema={uiSchema[propName]}
+                  errorSchema={errorSchema[propName]}
+                  idSchema={idSchema[propName]}
+                  formData={formData[propName]}
+                  onChange={this.onPropertyChange(propName)}
+                  onBlur={onBlur}
+                  registry={registry}
+                  disabled={disabled}
+                  readonly={readonly}/>
+              </div>
+            );
+          })}
+          <button className="usa-button-primary update-button" onClick={this.update}>Update</button>
+        </div>
+      </div>
+    );
+  }
+
+
+  getTitle = () => {
```

Comment:
I hate myself for commenting on this, but would it make sense to move `getTitle` above the `const title = this.getTitle();` on L89?

Reply:
Don't make no nevermind to me. Eslint is opinionated on how these functions get ordered sometimes, but I don't know that it is on this one.